internal/controller/sink.go

```diff
@@ -311,6 +312,13 @@ func (c *configSink) updateExtProcConfigMap(aiGatewayRoute *aigv1a1.AIGatewayRou
 			fc.Type = filterconfig.LLMRequestCostTypeTotalToken
 		case aigv1a1.LLMRequestCostTypeCEL:
 			fc.Type = filterconfig.LLMRequestCostTypeCELExpression
+			expr := *cost.CELExpression
+			// Sanity check the CEL expression.
+			_, err := llmcostcel.NewProgram(expr)
+			if err != nil {
+				return fmt.Errorf("invalid CEL expression: %w", err)
+			}
+			fc.CELExpression = expr
```

Comment:
Could we validate this in a webhook instead of controller as user is not going to get the error feedback if they specify the invalid cel expression ?

Reply:
even if that's possible, still we need to do the type validation.

Reply:
even if that's possible, still we need to do the type validation.

Reply:
let me see how much k8s API can validate a CEL expression (not sure even if it's doable)

Reply:
It is doable, need to setup a validating web hook to achieve this, we can do this post 0.1 though.

Reply:
hmm, i did a quick research but is it really possible to write a k8s validation to verify a string field content itself being a valid CEL expression?

Reply:
Essentially you add the same code in a webhook validation handler and then user can get the error feedback, thought we still need this code to deal with the standalone setup with configmap.

Reply:
ah ok i see not the CRD CEL validation, but instead k8s validation webhook. let's do that later after v0.1.0